assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/PieChart.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * PieChart component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { DATE_RANGE_OFFSET, STORE_NAME } from '../../../datastore/constants';
+import GoogleChart from '../../../../../components/GoogleChart';
+import { extractAnalyticsDataForPieChart } from '../../../util';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import PreviewBlock from '../../../../../components/PreviewBlock';
+
+const { useSelect } = Data;
+
+export default function PieChart() {
+	const {
+		hasFinishedResolution,
+		report,
+		error,
+	} = useSelect( ( select ) => {
```

Comment:
Looks like we already pull this report in the index.js file. So we can pass `report` object via props to this component and get rid of this useSelect hook.

Reply:
Updated.